src/tests/ftest/util/AgentUtils.py

```diff
@@ -127,31 +127,34 @@ def run_agent(basepath, server_list, client_list=None):
         sessions[client] = p
 
     # double check agent launched successfully
-    file_desc = sessions[client].stderr.fileno()
-    flags = fcntl.fcntl(file_desc, fcntl.F_GETFL)
-    fcntl.fcntl(file_desc, fcntl.F_SETFL, flags | os.O_NONBLOCK)
     timeout = 5
-    start_time = time.time()
-    result = 0
-    pattern = "Starting daos_agent"
-    expected_data = ""
-    while True:
-        output = ""
-        try:
-            output = sessions[client].stderr.read()
-        except IOError as excpn:
-            if excpn.errno != errno.EAGAIN:
-                raise excpn
-            continue
-        match = re.findall(pattern, output)
-        expected_data += output
-        result += len(match)
-        if not output or time.time() - start_time > timeout:
-            print("<AGENT>: {}".format(expected_data))
-            raise AgentFailed("DAOS Agent didn't start!")
-        break
-    print("<AGENT> agent started and took %s seconds to start" % \
-            (time.time() - start_time))
+    for client in client_list:
```

Comment:
It's worth keeping in mind that iterating serially over a list of clients like this is not going to scale very well.  It's unnoticeable for small numbers of clients, but as the number grows, the amount of time they are processed serially is going to take longer and longer.

Could use threading here to run many threads in parallel.

Reply:
I agree, though I am putting this into the category of "should have its own ticket". It might be a really fast ticket but still enough such that I'd rather it be its own patch than include it in bugfix patch.

Reply:
Indeed.